components/integrations/bots/bots.jsx

```diff
@@ -0,0 +1,206 @@
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import * as Utils from 'utils/utils.jsx';
+import BackstageList from 'components/backstage/components/backstage_list.jsx';
+import Constants from 'utils/constants.jsx';
+
+import Bot from './bot.jsx';
+
+export default class Bots extends React.PureComponent {
+    static propTypes = {
+
+        /**
+        *  Map from botUserId to bot.
+        */
+        bots: PropTypes.object.isRequired,
+
+        /**
+        *  Map from botUserId to accessTokens.
+        */
+        accessTokens: PropTypes.object.isRequired,
+
+        /**
+        *  Map from botUserId to owner.
+        */
+        owners: PropTypes.object.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+            * Ensure we have bot accounts
+            */
+            loadBots: PropTypes.func.isRequired,
+
+            /**
+            * Load access tokens for bot accounts
+            */
+            getUserAccessTokensForUser: PropTypes.func.isRequired,
+
+            /**
+            * Access token managment
+            */
+            createUserAccessToken: PropTypes.func.isRequired,
+            revokeUserAccessToken: PropTypes.func.isRequired,
+            enableUserAccessToken: PropTypes.func.isRequired,
+            disableUserAccessToken: PropTypes.func.isRequired,
+
+            /**
+            * Load owner of bot account
+            */
+            getUser: PropTypes.func.isRequired,
+
+            /**
+            * Disable a bot
+            */
+            disableBot: PropTypes.func.isRequired,
+
+            /**
+            * Enable a bot
+            */
+            enableBot: PropTypes.func.isRequired,
+        }),
+
+        /**
+        *  Only used for routing since backstage is team based.
+        */
+        team: PropTypes.object.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+        };
+    }
+
+    componentDidMount() {
+        this.props.actions.loadBots(
+            Constants.Integrations.START_PAGE_NUM,
+            Constants.Integrations.PAGE_SIZE
+        ).then(
+            (result) => {
+                if (result) {
+                    // We don't need to wait for this and we need to accept failure in the case where bot.owner_id is a plugin id
+                    result.data.map((bot) => this.props.actions.getUser(bot.owner_id));
+                    Promise.all(
+                        result.data.map((bot) => this.props.actions.getUserAccessTokensForUser(bot.user_id)).
+                            concat(result.data.map((bot) => this.props.actions.getUser(bot.user_id)))).
+                        then(() => {
+                            this.setState({loading: false});
+                        });
+                }
+            }
+        );
+    }
+
+    DisabledSection(props) {
```

Comment:
Minor nitpick, but if we want these to be consistent with other code, these methods would be named like `renderDisabledSection` and `renderEnabledSection` and then they'd just be called in the render method instead of using JSX like `<this.DisabledSection>`. That being said, I think I see what you're trying to go for with making them into functional components, so I'm 0/5 on changing them

Reply:
They actually need to be real components for the magic in `BackstageList` to work. That's why I did it like that.

Reply:
Oh, I see. I didn't realize that that's needed for `cloneElement`